src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1190,18 +1190,24 @@ public Component getTableCellRendererComponent(JTable table, Object value,
                 setIcon(icon);
 
             } else if (value instanceof Long) {
-                long len = ((Long) value) / 1024L;
+                long len = ((Long) value);
                 if (listViewWindowsStyle) {
+                    len /= 1024L;
                     text = MessageFormat.format(kiloByteString, len + 1);
                 } else if (len < 1024L) {
-                    text = MessageFormat.format(kiloByteString, (len == 0L) ? 1L : len);
+                    text = (len == 0L) ? 0 + " bytes" : len + " bytes";
```

Comment:
Is this necessary?
0 + " bytes" -->  "0 bytes"

Also, will this code work correctly for non-English locales?

Reply:
> Also, will this code work correctly for non-English locales?

It won't.

It has to use `MessageFormat` and it has to add a new string to the resource bundle.

Reply:
Dont think this len==0 check is needed...
One can use text = len + "bytes";

Also, this has i18n connection to it, so you need to use "byteString" similar to "kiloByteString", point it to 
UIManager.getString("FileChooser.fileSizeBytes", l); and add "FileChooser.fileSizeBytes" entry into 
share/classes/com/sun/swing/internal/plaf/basic/resources/basic.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_de.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_es.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_fr.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_it.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_ja.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_ko.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_pt_BR.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_sv.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_zh_CN.properties
share/classes/com/sun/swing/internal/plaf/basic/resources/basic_zh_TW.properties

I am not sure if need i18n team's intervention to update those resources as "bytes" can be translated to native in jp, zh
Maybe @naotoj can throw some light on that how to go about updating those resource file

Reply:
> > Also, will this code work correctly for non-English locales?
> 
> It won't.
> 
> It has to use `MessageFormat` and it has to add a new string to the resource bundle.

Yeah, I will check on this

Reply:
The code should employ `MessageFormat`. You only need to modify the English resource bundles, i.e., `basic.properties`, the rest will be taken care of by the g11n process.

Reply:
In this case, I would suggest to use straight number.

If you notice, the lengths are listed as {0} KB, {0} MB, {0} GB (there are no terabytes, I see), and "1023 bytes" would make the string too long and might case ellipses to appear.

Also, the threshold might be set at <= 999 instead of 1024.  The reason is group separator - the user might prefer to see 1,023 instead of 1023.

Ideally, there should be a pluggable policy (formatter) to format the lengths to the user's liking.

To summarize,
0...999 -> "0" ... "999"
1000 ... 1023 -> "1 KB"

Reply:
Another option is to use `java.text.CompactNumberFormat` which can handle those K/M/B suffixes nicely.

Reply:
Would java.text.CompactNumberFormat work for java 8?

Reply:
It is introduced in JDK12. If this has to be backported to 8, that's not an option.

Reply:
> It is introduced in JDK12. If this has to be backported to 8, that's not an option.

Is this fix going to be backported to 11 and 8?

Reply:
> The code should employ `MessageFormat`. You only need to modify the English resource bundles, i.e., `basic.properties`, the rest will be taken care of by the g11n process.

Ok, will update in basic.properties and check.

Reply:
@naotoj @aivanov-jdk @andy-goryachev-oracle Messageformat has been used to format the string to work correctly for non-english locale also. 
Modified the basic.properties file to handle byte values.